### PR TITLE
adds new method to convert counts to rank abundances using Table.transform

### PR DESCRIPTION
**Pull request by gregcaporaso**

### CODE REVIEW

**Comment by wasade** [comment]
:+1:, though can only review from phone. Looks good passing through. Thanks @gregcaporaso!

**Comment by gregcaporaso** [comment]
Thanks. 

To dos:
 - [ ] add test that includes passing a different ``method``

Other than that, should be good to go.

**Comment by wasade** [comment]
Is the sanity check of method then deferred to scipy? I think that's fine
-- I can't remember if there was a `Raises` section but if not, it may be
good to include a mention of what might arise with a bad `method` or
perhaps in `Notes`

File: biom/cli/table_summarizer.py [hunk 1]
```diff
@@ -101,11 +108,14 @@ def _summarize_table(table, qualitative=False, observations=False):
     else:
         lines.append('Counts/sample summary:')
 
-    lines.append(' Min: %r' % min_counts)
-    lines.append(' Max: %r' % max_counts)
-    lines.append(' Median: %1.3f' % median_counts)
-    lines.append(' Mean: %1.3f' % mean_counts)
-    lines.append(' Std. dev.: %1.3f' % std(counts_per_sample_values))
+    lines.append(' Min: ' + locale.format('%1.3f', min_counts, grouping=True))
```
Comment (gregcaporaso) [inline biom/cli/table_summarizer.py]: Sadly ``%r`` doesn't work with ``locale.format``.

**Comment by gregcaporaso** [comment]
This is ready for final review.

Note that I ended up addressing #645 here too - see screenshot of example output in that issue.

File: biom/table.py [hunk 2]
```diff
@@ -2722,6 +2723,60 @@ def transform(self, f, axis='sample', inplace=True):
 
         return table
 
+    def rankdata(self, axis='sample', inplace=True, method='average'):
+        """Convert counts to rank abundances from smallest to largest
```
Comment (wasade) [inline biom/table.py]: It isn't always the case that the data contained are counts as they may be relative abundances.

File: biom/table.py [hunk 3]
```diff
@@ -2722,6 +2723,60 @@ def transform(self, f, axis='sample', inplace=True):
...
+
+        Parameters
+        ----------
+        axis : {'sample', 'observation'}, optional
+            The axis to use for normalization
```
Comment (wasade) [inline biom/table.py]: Can you add a period please?

File: biom/table.py [hunk 4]
```diff
@@ -2722,6 +2723,60 @@ def transform(self, f, axis='sample', inplace=True):
...
+        inplace : bool, optional
+            Defaults to ``True``. If ``True``, performs the normalization in
+            place. Otherwise, returns a new table with the noramlization
```
Comment (wasade) [inline biom/table.py]: misspelling of normalization

File: biom/table.py [hunk 5]
```diff
@@ -2722,6 +2723,60 @@ def transform(self, f, axis='sample', inplace=True):
...
+            applied.
+        method : str, optional
+            The method for handling ties in counts. This can be any valid
+            string that can be passed to `scipy.stats.rankdata`.
+
+        Returns
+        -------
+        biom.Table
+            The rank-abundance-transformed table.
```
Comment (wasade) [inline biom/table.py]: is that hyphenation correct?

Comment (gregcaporaso) [inline biom/table.py]: Yesh, b/c both terms are modifying ``transformed``.

File: ChangeLog.md [hunk 6]
```diff
@@ -6,6 +6,11 @@ biom 2.1.5-dev
 
 Changes since biom 2.1.5 go here.
 
+New Features:
+* `Table.rankdata` has been added to convert counts to ranked abundances on
```
Comment (wasade) [inline ChangeLog.md]: same comment here re: counts

File: ChangeLog.md [hunk 7]
```diff
@@ -6,6 +6,11 @@ biom 2.1.5-dev
...
+  either axis. See [issue #645](https://github.com/biocore/biom-format/issues/639).
+* Format of numbers in ``biom summarize-table`` output is now more readable. See [issue #679](https://github.com/biocore/biom-format/issues/679).
```
Comment (wasade) [inline ChangeLog.md]: "more readable" -> "localized"

**Comment by wasade** [comment]
Just a few comments

**Comment by gregcaporaso** [comment]
Addressed your comments @wasade.

**Comment by wasade** [comment]
Thanks!

**Comment by gregcaporaso** [comment]
Awesome, thank you!

**Pull request merged**